tracer/test/Datadog.Trace.IntegrationTests/SpanTagTests.cs

```diff
@@ -0,0 +1,74 @@
+// <copyright file="SpanTagTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Sampling;
+using Xunit;
+
+namespace Datadog.Trace.IntegrationTests
+{
+    public class SpanTagTests
+    {
+        private readonly Tracer _tracer;
+
+        public SpanTagTests()
+        {
+            var matchAllRule = "[{\"service\":\"*\", \"name\":\"*\", \"sample_rate\":1.0, \"max_per_second\":1000.0}]";
+            var settings = new TracerSettings() { SpanSamplingRules = matchAllRule };
+            // a spanSampler should be generated due to the TracerSettings containing the SpanSamplingRules
+            _tracer = new Tracer(settings, null, sampler: null, spanSampler: null, scopeManager: null, statsd: null);
+        }
+
+        [Fact]
+        public void SpanSampler_ShouldAddTags_OnSpanClose_ForDroppedTrace()
+        {
+            var traceContext = new TraceContext(_tracer);
+            var expectedRuleRate = "1";
+            var expectedMaxPerSecond = "1000";
+            var expectedSamplingMechanism = "8";
+            var span = new Span(new SpanContext(5, 6, null, serviceName: "service"), DateTimeOffset.Now) { OperationName = "operation" };
+            var span2 = new Span(new SpanContext(5, 7, null, serviceName: "service"), DateTimeOffset.Now) { OperationName = "operation" };
+            // mechanism not important, but we've decided to drop the trace
+            traceContext.SetSamplingPriority(SamplingPriorityValues.UserReject, SamplingMechanism.Manual);
+            traceContext.AddSpan(span2);
+            traceContext.CloseSpan(span2);
+
+            traceContext.AddSpan(span);
+            traceContext.CloseSpan(span);
```

Comment:
It shouldn't change anything, but the common scenario is more:
- AddSpan
- AddSpan
- CloseSpan
- CloseSpan

As we have one root span in traces usually

Reply:
Makes sense to me, also realize that I had those spans swapped around so fixed both in 02a0060072eb0916686095ffc91ebef5792fd7e7